TestingGrounds/TestingGrounds/Nodes.hpp

```diff
@@ -28,4 +28,37 @@ struct SingleNode
     }
     uint32_t value;
     SingleNode* next;
+};
+
+template <typename K, typename V>
+struct HashNode
+{
+    HashNode(const K &key, const V &value):
+        key(key),
+        value(value)
+    {}
+    ~HashNode()
+    {
+        key = NULL;
+        value = NULL;
+    }
+    K key;
+    V value;
+
+    K getKey( void )
+    {
+        return key;
+    }
+    V getValue( void )
+    {
+        return value;
+    }
+    void setKey( K key )
```

Comment:
There are some wonky things going on here with mixing `.` and `->`.

Reply:
Un-donkied